Feature/actor ttl

chinzhiweiblank opened the pull request:
# Description

_Please explain the changes you've made_

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

dependabot[bot] committed:
Bump mypy from 0.910 to 0.920 (#314)

Bumps [mypy](https://github.com/python/mypy) from 0.910 to 0.920.
- [Release notes](https://github.com/python/mypy/releases)
- [Commits](https://github.com/python/mypy/compare/v0.910...v0.920)

---
updated-dependencies:
- dependency-name: mypy
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

chinzhiweiblank committed:
Initial commit of ttl

Signed-off-by: chinzhiweiblank <chinzhiweiblank@gmail.com>
Signed-off-by: root <root@DESKTOP-C475BU9.localdomain>
Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

chinzhiweiblank committed:
Fix dictionary serialization order

Signed-off-by: chinzhiweiblank <chinzhiweiblank@gmail.com>
Signed-off-by: root <root@DESKTOP-C475BU9.localdomain>
Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

chinzhiweiblank committed:
Fix linting

Signed-off-by: root <root@DESKTOP-C475BU9.localdomain>
Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

Commit:
Remove ttl from examples

Signed-off-by: root <root@DESKTOP-C475BU9.localdomain>
Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

Commit:
Add back ttl

Signed-off-by: root <root@DESKTOP-C475BU9.localdomain>
Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

Commit:
Add ttl into receive_reminder

Signed-off-by: root <root@DESKTOP-C475BU9.localdomain>
Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

dependabot[bot] committed:
Bump mypy from 0.920 to 0.930 (#319)

Bumps [mypy](https://github.com/python/mypy) from 0.920 to 0.930.
- [Release notes](https://github.com/python/mypy/releases)
- [Commits](https://github.com/python/mypy/compare/v0.920...v0.930)

---
updated-dependencies:
- dependency-name: mypy
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

willtsai committed:
Add DCO requirements and updated code of conduct (#321)

Issue reference: dapr/docs#2039

Signed-off-by: Will <william.wl.tsai@gmail.com>

Co-authored-by: Charlie Stanley <charlie.stanley@microsoft.com>

dependabot[bot] committed:
Bump tox from 3.24.4 to 3.24.5 (#322)

Bumps [tox](https://github.com/tox-dev/tox) from 3.24.4 to 3.24.5.
- [Release notes](https://github.com/tox-dev/tox/releases)
- [Changelog](https://github.com/tox-dev/tox/blob/master/docs/changelog.rst)
- [Commits](https://github.com/tox-dev/tox/compare/3.24.4...3.24.5)

---
updated-dependencies:
- dependency-name: tox
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

amulyavarote committed:
Added get configuration method for Dapr config API (#323)

* Added get configuration method for Dapr config API

Signed-off-by: Amulya Varote <amulyavarote@Ruban-laptop.fareast.corp.microsoft.com>

* Fixed linting issues

Signed-off-by: Amulya Varote <amulyavarote@Ruban-laptop.fareast.corp.microsoft.com>

* Fixed github actions build error

Signed-off-by: Amulya Varote <amulyavarote@SHRISU-YOGA.redmond.corp.microsoft.com>

* Changes based on the review comments - 1

Signed-off-by: Amulya Varote <amulyavarote@SHRISU-YOGA.redmond.corp.microsoft.com>

* Changes based on the review comments - 2

Signed-off-by: Amulya Varote <amulyavarote@SHRISU-YOGA.redmond.corp.microsoft.com>

* Corrected the comment

Signed-off-by: Amulya Varote <amulyavarote@Amulyas-MacBook-Pro.local>

Co-authored-by: Amulya Varote <amulyavarote@Ruban-laptop.fareast.corp.microsoft.com>
Co-authored-by: Amulya Varote <amulyavarote@SHRISU-YOGA.redmond.corp.microsoft.com>
Co-authored-by: Amulya Varote <amulyavarote@Amulyas-MacBook-Pro.local>

berndverst commented:
@chinzhiweiblank can you make the changes, but then actually close this Pull request and open it again `dapr/python-sdk:master` instead?

I just rebased your branch `feature/actor_ttl` against the latest protos from my branch for PR https://github.com/dapr/python-sdk/pull/326

It would be great if you could update the docs strings in your branch. Then close this PR and reopen the PR against the upstream repo's master branch.

Thanks for the help!

berndverst committed:
Update Latest Dapr Protos (#326)

Signed-off-by: GitHub <noreply@github.com>
Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

dependabot[bot] committed:
Bump mypy from 0.930 to 0.931 (#329)

Bumps [mypy](https://github.com/python/mypy) from 0.930 to 0.931.
- [Release notes](https://github.com/python/mypy/releases)
- [Commits](https://github.com/python/mypy/compare/v0.930...v0.931)

---
updated-dependencies:
- dependency-name: mypy
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Charlie Stanley <charlie.stanley@microsoft.com>

chinzhiweiblank committed:
Add docstrings

Signed-off-by: chinzhiweiblank <chinzhiweiblank@gmail.com>
Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

paulyuk committed:
Add deprecation warnings for unused gRPC metadata (#335)

* Add deprecation warnings for unused gRPC metadata

Signed-off-by: Paul Yuknewicz <paulyuk@microsoft.com>

amulyavarote committed:
Examples modified (#337)

Signed-off-by: Amulya Varote <amulyavarote@Amulyas-MacBook-Pro.local>

chinzhiweiblank committed:
Add timedelta fix (#327)

* Add timedelta fix

Signed-off-by: chinzhiweiblank <chinzhiweiblank@gmail.com>

berndverst committed:
Fix incorrect license configuration (#338)

* fix incorrect license configuration

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Update supported Python versions

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

dependabot[bot] committed:
Bump protobuf from 3.19.1 to 3.19.3 (#332)

* Bump protobuf from 3.19.1 to 3.19.3

Bumps [protobuf](https://github.com/protocolbuffers/protobuf) from 3.19.1 to 3.19.3.
- [Release notes](https://github.com/protocolbuffers/protobuf/releases)
- [Changelog](https://github.com/protocolbuffers/protobuf/blob/master/generate_changelog.py)
- [Commits](https://github.com/protocolbuffers/protobuf/commits)

---
updated-dependencies:
- dependency-name: protobuf
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

* setup.cfg: Bump protobuf from 3.19.1 to 3.19.3

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>
Co-authored-by: Charlie Stanley <charlie.stanley@microsoft.com>

amulyavarote committed:
Added a warning message (#339)

Signed-off-by: Amulya Varote <amulyavarote@Amulyas-MacBook-Pro.local>

berndverst committed:
Adds support for State Store Query Alpha API (#325)

* Query State Store Tests

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* query API mechanical markdown

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* fix type

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* fix linter issues

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Update dapr/clients/grpc/_response.py

Co-authored-by: Paul Yuknewicz <paulyuk@microsoft.com>
Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Update dapr/clients/grpc/_response.py

Co-authored-by: Paul Yuknewicz <paulyuk@microsoft.com>
Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Update dapr/clients/grpc/_response.py

Co-authored-by: Paul Yuknewicz <paulyuk@microsoft.com>
Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Update dapr/clients/grpc/_response.py

Co-authored-by: Paul Yuknewicz <paulyuk@microsoft.com>
Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Update dapr/clients/grpc/_response.py

Co-authored-by: Paul Yuknewicz <paulyuk@microsoft.com>
Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Update examples/state_store_query/README.md

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Query API name change and alpha warning

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Update warning message

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

Co-authored-by: Paul Yuknewicz <paulyuk@microsoft.com>

amulyavarote committed:
Added config API documentation (#345)

Signed-off-by: Amulya Varote <amulyavarote@Amulyas-MacBook-Pro.local>

Co-authored-by: Amulya Varote <amulyavarote@Amulyas-MacBook-Pro.local>

berndverst committed:
add state store query api to docs (#348)

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

chinzhiweiblank committed:
Test ttl

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

chinzhiweiblank committed:
Test with ttl

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

berndverst committed:
Add PubSub subscriptions to Flask Dapr (#351)

* Add PubSub subscriptions to Flask Dapr

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

dependabot[bot] committed:
Bump protobuf from 3.19.3 to 3.19.4 (#352)

Bumps [protobuf](https://github.com/protocolbuffers/protobuf) from 3.19.3 to 3.19.4.
- [Release notes](https://github.com/protocolbuffers/protobuf/releases)
- [Changelog](https://github.com/protocolbuffers/protobuf/blob/master/generate_changelog.py)
- [Commits](https://github.com/protocolbuffers/protobuf/compare/v3.19.3...v3.19.4)

---
updated-dependencies:
- dependency-name: protobuf
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

dependabot[bot] committed:
Bump azure/setup-helm from 1 to 2.0 (#349)

Bumps [azure/setup-helm](https://github.com/azure/setup-helm) from 1 to 2.0.
- [Release notes](https://github.com/azure/setup-helm/releases)
- [Commits](https://github.com/azure/setup-helm/compare/v1...v2.0)

---
updated-dependencies:
- dependency-name: azure/setup-helm
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

berndverst committed:
Change default value of soon to be removed grpc metadata (#356)

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

psychomelet committed:
fix timer context (#330)

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

dependabot[bot] committed:
Bump actions/setup-python from 2 to 3 (#360)

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 2 to 3.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

berndverst committed:
Adds Dapr Bot & Fossa Dependency License Scan (#362)

* Adds Fossa Scan

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Add Dapr Bot

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Fix linter errors for automerge script from dapr/dapr

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Use FOSSA Push Only token in workflow

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

dependabot[bot] committed:
Bump actions/checkout from 2 to 3 (#363)

Bumps [actions/checkout](https://github.com/actions/checkout) from 2 to 3.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump actions/github-script from 1 to 6 (#365)

Bumps [actions/github-script](https://github.com/actions/github-script) from 1 to 6.
- [Release notes](https://github.com/actions/github-script/releases)
- [Commits](https://github.com/actions/github-script/compare/v1...v6)

---
updated-dependencies:
- dependency-name: actions/github-script
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump actions/stale from 3.0.14 to 5 (#364)

Bumps [actions/stale](https://github.com/actions/stale) from 3.0.14 to 5.
- [Release notes](https://github.com/actions/stale/releases)
- [Changelog](https://github.com/actions/stale/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/stale/compare/v3.0.14...v5)

---
updated-dependencies:
- dependency-name: actions/stale
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
Add topic event response support to GRPC ext client (#341)

* Add missing topic event and status to grpc client

This add the already existing objects TopicEventResponse and
TopicEventResponseStatus from protobuf to the grpc client

Signed-off-by: Renato L. de F. Cunha <renato.cunha@microsoft.com>
Signed-off-by: GitHub <noreply@github.com>

* export TopicEventResponse from grpc-ext package

Signed-off-by: Renato L. de F. Cunha <renato.cunha@microsoft.com>
Signed-off-by: GitHub <noreply@github.com>

* add TopicEventResponse support to servicier

Signed-off-by: Renato L. de F. Cunha <renato.cunha@microsoft.com>
Signed-off-by: GitHub <noreply@github.com>

* add test to verify TopicEventResponses are returned

Signed-off-by: Renato L. de F. Cunha <renato.cunha@microsoft.com>
Signed-off-by: GitHub <noreply@github.com>

* add TopicEventResponse example

Signed-off-by: Renato L. de F. Cunha <renato.cunha@microsoft.com>
Signed-off-by: GitHub <noreply@github.com>

* Document how to use TopicEventResponse

TopicEventResponse instructs dapr on what to do after
a message is received.

Signed-off-by: Renato L. de F. Cunha <renato.cunha@microsoft.com>
Signed-off-by: GitHub <noreply@github.com>

* Added config API documentation (#345)

Signed-off-by: Amulya Varote <amulyavarote@Amulyas-MacBook-Pro.local>

Co-authored-by: Amulya Varote <amulyavarote@Amulyas-MacBook-Pro.local>
Signed-off-by: GitHub <noreply@github.com>

* add state store query api to docs (#348)

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>
Signed-off-by: GitHub <noreply@github.com>

* Add PubSub subscriptions to Flask Dapr (#351)

* Add PubSub subscriptions to Flask Dapr

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>
Signed-off-by: GitHub <noreply@github.com>

* Bump protobuf from 3.19.3 to 3.19.4 (#352)

Bumps [protobuf](https://github.com/protocolbuffers/protobuf) from 3.19.3 to 3.19.4.
- [Release notes](https://github.com/protocolbuffers/protobuf/releases)
- [Changelog](https://github.com/protocolbuffers/protobuf/blob/master/generate_changelog.py)
- [Commits](https://github.com/protocolbuffers/protobuf/compare/v3.19.3...v3.19.4)

---
updated-dependencies:
- dependency-name: protobuf
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>
Signed-off-by: GitHub <noreply@github.com>

* Bump azure/setup-helm from 1 to 2.0 (#349)

Bumps [azure/setup-helm](https://github.com/azure/setup-helm) from 1 to 2.0.
- [Release notes](https://github.com/azure/setup-helm/releases)
- [Commits](https://github.com/azure/setup-helm/compare/v1...v2.0)

---
updated-dependencies:
- dependency-name: azure/setup-helm
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>
Signed-off-by: GitHub <noreply@github.com>

* Add tests for creating TopicEventResponse

Signed-off-by: Renato L. de F. Cunha <renato.cunha@microsoft.com>

* Make tests pass

Signed-off-by: Renato L. de F. Cunha <renato.cunha@microsoft.com>

* fix linting errors in test file

Signed-off-by: Renato Luiz de Freitas Cunha <renato.cunha@microsoft.com>

* Update examples/pubsub-simple to show how to retry

The pubsub-simple example now tells dapr to retry delivery on the first
message the subscriber receives.

The message is logged to stdout by dapr and we capture that with mm.

Since we are only retrying the first message, stdout will show the old
output, plus the first line repeated, and the dapr log message.

Signed-off-by: Renato Luiz de Freitas Cunha <renato.cunha@microsoft.com>

* remove line for redelivered message

Apparently, not all dapr implementations are guaranteed to redeliver
messages right away, as mentioned in the comments of PR #341.

Signed-off-by: Renato Luiz de Freitas Cunha <renato.cunha@microsoft.com>

* fix README by not saying retried msgs are printed twice

Signed-off-by: Renato Luiz de Freitas Cunha <renato.cunha@microsoft.com>

Co-authored-by: amulyavarote <91495930+amulyavarote@users.noreply.github.com>
Co-authored-by: Amulya Varote <amulyavarote@Amulyas-MacBook-Pro.local>
Co-authored-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

berndverst committed:
Switch to Dapr org Fossa credential (#367)

* Update fossa.yaml

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Fossa shield

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

dependabot[bot] committed:
Bump mypy from 0.931 to 0.940 (#366)

Bumps [mypy](https://github.com/python/mypy) from 0.931 to 0.940.
- [Release notes](https://github.com/python/mypy/releases)
- [Commits](https://github.com/python/mypy/compare/v0.931...v0.940)

---
updated-dependencies:
- dependency-name: mypy
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

dependabot[bot] committed:
Bump mypy from 0.940 to 0.941 (#368)

Bumps [mypy](https://github.com/python/mypy) from 0.940 to 0.941.
- [Release notes](https://github.com/python/mypy/releases)
- [Commits](https://github.com/python/mypy/compare/v0.940...v0.941)

---
updated-dependencies:
- dependency-name: mypy
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

berndverst committed:
fix linter

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

berndverst committed:
Make ttl optional

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

dependabot[bot] committed:
Bump mypy from 0.941 to 0.942 (#374)

* Bump mypy from 0.941 to 0.942

Bumps [mypy](https://github.com/python/mypy) from 0.941 to 0.942.
- [Release notes](https://github.com/python/mypy/releases)
- [Commits](https://github.com/python/mypy/compare/v0.941...v0.942)

---
updated-dependencies:
- dependency-name: mypy
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* Fix broken docs link

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

berndverst committed:
Update examples for new state store query syntax (#357)

* Update examples for new state store query syntax

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Update query example output

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

Co-authored-by: Mukundan Sundararajan <65565396+mukundansundar@users.noreply.github.com>

berndverst committed:
Merge branch 'master' into feature/actor_ttl

berndverst committed:
Allow configuring of max gRPC message length (#375)

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

berndverst committed:
Always use existing asyncio event loop (#376)

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

amulyavarote committed:
Added subscribe config API support (#369)

* Added subscribe config API support

Signed-off-by: Amulya Varote <amulyavarote@QTM-SWATHIKIL-1.redmond.corp.microsoft.com>
Signed-off-by: Amulya Varote <amulyavarote@microsoft.com>

* Fixed subscribe feature

Signed-off-by: Amulya Varote <amulyavarote@Amulyas-MacBook-Pro.local>
Signed-off-by: Amulya Varote <amulyavarote@microsoft.com>

* Fixing build failures

Signed-off-by: Amulya Varote <amulyavarote@Amulyas-MacBook-Pro.local>
Signed-off-by: Amulya Varote <amulyavarote@microsoft.com>

* Changes based on the review comments

Signed-off-by: Amulya Varote <amulyavarote@Amulyas-MacBook-Pro.local>
Signed-off-by: Amulya Varote <amulyavarote@microsoft.com>

* Changes based on the review comments

Signed-off-by: Amulya Varote <amulyavarote@QTM-SWATHIKIL-1.redmond.corp.microsoft.com>
Signed-off-by: Amulya Varote <amulyavarote@microsoft.com>

* Fixed mechanical markdown tests

Signed-off-by: Amulya Varote <amulyavarote@microsoft.com>

* Changed sleep time

Signed-off-by: Amulya Varote <amulyavarote@microsoft.com>

* Changed sleep time

Signed-off-by: Amulya Varote <amulyavarote@microsoft.com>

* Validation in the same step

Signed-off-by: Amulya Varote <amulyavarote@microsoft.com>

* Validation in the same step

Signed-off-by: Amulya Varote <amulyavarote@microsoft.com>

* Combining examples

Signed-off-by: Amulya Varote <amulyavarote@microsoft.com>

* Removed other example file

Signed-off-by: Amulya Varote <amulyavarote@microsoft.com>

* Removed subscribe line

Signed-off-by: Amulya Varote <amulyavarote@microsoft.com>

* Fixed external link validation

Signed-off-by: Amulya Varote <amulyavarote@microsoft.com>

* Added back subscribe statement in readme

Signed-off-by: Amulya Varote <amulyavarote@microsoft.com>

* Changes based on the review comments

Signed-off-by: Amulya Varote <amulyavarote@microsoft.com>

* Removed virtual env

Signed-off-by: Amulya Varote <amulyavarote@microsoft.com>

Co-authored-by: Amulya Varote <amulyavarote@Amulyas-MacBook-Pro.local>
Co-authored-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

berndverst committed:
Use correct linters in devcontainer and add manual linter instructions (#373)

* Add linter instructions

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Enable correct linters in devcontainer

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Fix broken docs link

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

yaron2 committed:
Merge branch 'master' into feature/actor_ttl

berndverst committed:
make ttl optional

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>
